sql/parser/builtins.go

```diff
@@ -382,8 +382,8 @@ var builtins = map[string][]builtin{
 			types:      typeList{timestampType},
 			returnType: DummyInterval,
 			impure:     true,
```

Comment:
Yeah, this is a bit ambiguous. If you look at how we use `impure` currently, we only care if the expression is a constant, and we only care about that for the purpose of expression simplification.

I think it would be fair to say that purity is statement-scoped, but the update comment should mention the specifics of how `impure` is accessed.